flume-ng-channels/flume-file-channel/src/main/java/org/apache/flume/channel/file/FileChannel.java

```diff
@@ -503,6 +511,7 @@ protected void doPut(Event event) throws InterruptedException {
         queue.addWithoutCommit(ptr, transactionID);
         success = true;
       } catch (IOException e) {
+        channelCounter.incrementEventPutErrorCount();
```

Comment:
Don't we want to mark the channel unhealthy here too? And elsewhere that we increment event take / put error count as well as checkpoint write failure?